### PR TITLE
[cryptotest] refactor `parse_rsp` and clean up test vector build script

### PR DESCRIPTION
**Pull request by RyanTorok**
Performs a massive overhaul (and breaking API change) of the `parse_rsp` function in the cryptotest framework's utility library for test vector parsers.

The heterogeneous formatting styles of the test vector files meant that the caller of `parse_rsp` was still having to do a lot of heavy lifting to get the data into the shape they wanted, and some test vectors were impossible to support using the previous version, such as AES-GCM and DRBG, which declare multiple of the same variable in the same test vector. Likewise, this PR is largely a response to the issues that came up during #21617 .

The new API does the following:

- Rather than sort the test vectors into sections based on the headers that looked like `[header]`, the test vectors are instead returned in a single long array. Headers are now interpreted in one of the following three ways:
    - If the header looks like `[key = value]`, then the key/value pair is simply inserted like a normal variable into every test vector that follows, until another value is set for the same key.
    - If the header looks like `[comma, separated, values]`, then the field `test_vec["section_name"]` is set to an array of the comma-separated strings, where `test_vec` is the test vector.
    - Otherwise, `test_vec["section_name"]` is set to a string equal to the entire text between the `[]`.
- Additionally, the API allows the caller to specify _persistent variables_, which flag the group of lines containing key-value pairs as variables that should be applied to multiple test cases, rather than having the group of lines being treated as a test case itself.
    - To specify that a variable is persistent, pass its name into the API like this: `parse_rsp(filename, ["n"])`. Then, any instance of the variable `n` will apply to all following test cases until another value is assigned to it.
    - Marking any variable in a line group as persistent will cause the group not to be added as a test case, but only the specified variables will be added to the following test vectors.
    - Persistent variables are treated the same as key/value pairs in headers, where the value will be applied to all test vectors until another value is assigned for the same key.
    - For an example of why this behavior is needed / how it works, check out how the variables `n`, `p`, and `q` are specified in the [NIST RSA test vectors](https://csrc.nist.gov/CSRC/media/Projects/Cryptographic-Algorithm-Validation-Program/documents/components/RSA2SP1testvectors.zip).
- Blank lines and headers are considered delimiters between line groups / test cases.
- If a key occurs more than once in a line group, all of the values will be included in the test vector in an array.
    - This also applies to caller-specified persistent variables, but not headers.

The implementation of the new semantics was complex enough to warrant its own set of unit tests, which are included with this PR.

Along for the ride are also a few minor changes to the test vector build script (e.g. changing `genrule` to `run_binary`).

### CODE REVIEW

**Review by sadmansakib-git — COMMENTED** [review]
Looks good to me. As this change is quite substantial, I would like to defer to @jadephilipoom or @milesdai for the final decision.

**Review by milesdai — APPROVED** [review]
Thanks Ryan! I have a few small nits, but this LGTM!

File: sw/host/cryptotest/testvectors/parsers/cryptotest_util_test.py
Comment (RyanTorok) [reply sw/host/cryptotest/testvectors/parsers/cryptotest_util_test.py:19]: Much simpler. thanks!

Comment (milesdai) [reply sw/host/cryptotest/testvectors/parsers/cryptotest_util_test.py:19]: Ah I was thinking we can probably just omit this `checkResult` function entirely and call `t.assertEqual` directly on the two lists in the test case without checking anything else. I don't think there's a need to check the equality of the dictionaries explicitly since Python defines equality of lists as equality of all the corresponding elements and the equality check already handles the length checking.

Comment (RyanTorok) [reply sw/host/cryptotest/testvectors/parsers/cryptotest_util_test.py:19]: I was concerned about rejecting an implementation that returned the correct test vectors in the wrong order, but that really can't happen. I think we can simplify this even more.

Comment (RyanTorok) [reply sw/host/cryptotest/testvectors/parsers/cryptotest_util_test.py]: Fixed

File: sw/host/cryptotest/testvectors/parsers/cryptotest_util.py
Comment (RyanTorok) [reply sw/host/cryptotest/testvectors/parsers/cryptotest_util.py]: Good catch. Making `kv` contain stripped values was a quick final change to allow the tests to pass, and I didn't catch that it made the later calls redundant.

**Pull request merged**
